lib/core/domain/repository/repository.dart

```diff
@@ -0,0 +1,17 @@
+import 'package:flutter/material.dart';
+import 'package:twake/models/globals/globals.dart';
+
+abstract class RepositoryHelper {
+  navigatorDatasource({
+    VoidCallback? onLocal,
+    VoidCallback? onRemote,
+  }) {
+    if (!Globals.instance.isNetworkConnected) {
+      onLocal?.call();
+      return;
+    }
+    onRemote?.call();
```

Comment:
IMO, it is not generic enough. For example: when we need to first: fetch data from local, in parallel, fetch the data with network to update.